lib/backward-compatilibity-layer.ts

```diff
@@ -0,0 +1,127 @@
+import {
```

Comment:
The philosophy behind introducing the backward compatibility layer was to avoid introducing breaking changes (in options interfaces). In the next major release, we'll simply remove the compatibility layer BUT features like `uiHooks` should IMO still keep working as expected @flamewow (otherwise we won't be able to guarantee 100% feature parity)

Reply:
hooks are the feature of fastify, i.e. developers won't be able to simply swap fastify to express adapter in a NestFactory.create method params and expect it to work the same way.

Initial idea was to achieve behavior like "upon swapping adapter dev should be able to get the same behavior from nestjs/swagger as it was with a different adapter without any modifications to swagger configuration" in a long run.
Otherwise, it's not truly platform-independent, the only thing we achieved is we got rid of extra dependency (express-swagger-ui, fastify-swagger).

Do you want to keep that uiHooks as an exclusive fastify-only feature in a long run, or to come up with a universal solution that would work for express & fastify alike?

Reply:
@flamewow can you run some tests with adding basic auth in front of Swagger UI (for fastify) using a simple middleware? Perhaps with this "universal" approach `uiHooks` can be swapped with something else (middleware-based approach)